### PR TITLE
cam6_3_023: fix for mpas regression failure (issue #389) and MPAS makefile update for NUOPC

### LINKED ISSUE #389: MPAS regression test failure for cam6_3_020
**FAIL ERC_D_Ln9.mpasa120z32_mpasa120.FKESSLER.cheyenne_intel.cam-outfrq3s_usecase COMPARE_base_rest**

CAM6_3_020 inadvertently included a regression test failure.   The comparison of the initial files produced from the base and restart run showed differences in some of the grid description fields. All other history files were BFB.
According to the cprnc.out file, 4 fields are different: edgesOnCell, edgesOnEdge, cellsOnCell, verticesOnCell.
The restart and initial files are only opened in CAM and the resulting filehandles are passed to the MPAS dycore to complete the writing of restart information.

### PR DESCRIPTION
**Pull request by jtruesdal**
New MPAS external hash to fix MPAS base/rest regression failure introduced in cam6_3_020.  Updated MPAS build routines to allow MPAS dycore library to link with the CESM ESMF library.  This change completes NUOPC compatibility for MPAS.

closes #389

### CODE REVIEW

File: Externals_CAM.cfg [hunk 1]
```diff
@@ -59,7 +59,7 @@ local_path = src/dynamics/mpas/dycore
 protocol = git
 repo_url = https://github.com/MPAS-Dev/MPAS-Model.git
 sparse = ../.mpas_sparse_checkout
-hash = 224740a
+hash = 8fb89189
```
Comment (cacraigucar) [inline Externals_CAM.cfg:62]: I assume that whatever "magic" that was being done for the previous hash to make sure it does not get deleted will be done for this hash as well.

**Review by cacraigucar — APPROVED** [review]

File: Externals_CAM.cfg (hunk 1 above)
Comment (jtruesdal) [reply Externals_CAM.cfg:62]: Yes.  Michael was able to show that he can make all of these hashes available from the main  MPAS development branch once everything is merged up.  Hopefully within the next 30 days.

File: doc/ChangeLog [hunk 2]
```diff
@@ -1,4 +1,66 @@
 ===============================================================
+Tag name: cam6_3_023
+Originator(s): jet, duda
+Date: 23 Jun 2021
+One-line Summary: Fix MPAS base/rest regression test failure (issue #389) allow MPAS NUOPC build
+Github PR URLs:
```
Comment (gold2718) [inline doc/ChangeLog:6]: This can be updated now: https://github.com/ESCOMP/CAM/pull/391

File: doc/ChangeLog [hunk 3]
```diff
@@ -1,4 +1,66 @@
...
+Code reviewed by: CAM SEs
+
+List all files eliminated:
+
+List all files added and what they do:
+
+List all existing files that have been modified, and describe the changes:
+
+M	Externals_CAM.cfg b/Externals_CAM.cfg
+ - fix base/rest regression test failure
```
Comment (gold2718) [inline doc/ChangeLog:35]: The first line looks like it has extra stuff. Also, please note that the MPAS external is what was updated here.

**Review by gold2718 — APPROVED** [review]
Please make a couple of adjustments to ChangeLog.

**Review by peverwhee — APPROVED** [review]

File: doc/ChangeLog (hunk 2 above)
Comment (jtruesdal) [reply doc/ChangeLog:6]: Will do.

File: doc/ChangeLog (hunk 3 above)
Comment (jtruesdal) [reply doc/ChangeLog:35]: Noted.  I'll push after I get Jesse's review

**Pull request merged**